src/evaluator/evaluator/adapter.py

```diff
@@ -0,0 +1,37 @@
+from sensor_msgs.msg import PointCloud2
+import message_filters
+import datetime
+
+class Adapter:
+    """
+    Class for subscribing to a point cloud topic and synchronizing messages with a ROS2 node.
+    """
+
+    def __init__(self, node, point_cloud_topic):
+        """
+        Initializes the Adapter.
+
+        Args:
+            node (Node): ROS2 node instance.
+            point_cloud_topic (str): Topic for point cloud data.
+        """
+        self.node = node
```

Comment:
You probably don't know about this, but you can use types in Python, and it makes it a lot better:
```suggestion
        self.node : rclpy.node.Node = node # or just 'Node', depending on import
```
This will make the IDE be able to recognize improper typings right away